utility/get_ip_addrs.py

```diff
@@ -0,0 +1,54 @@
+""" Look at Ceph storage pods, return ip-addresses """
+import subprocess
+import shlex
+from ocs.defaults import ROOK_CLUSTER_NAMESPACE
+from ocs import api_client as ac
+
+
+def get_ip_addrs_for_ceph_type(pod_type, namespace=ROOK_CLUSTER_NAMESPACE):
+    """
+    Get the ip address for all pods of a given type.
+
+    Args:
+        pod_type (str): pod type (ceph-mon, ceph-mgr, ceph-osd)
+        namespace (str): namespace in -n field of oc commands
+                         Defaults to ROOK_CLUSTER_NAMESPACE
+
+    Returns:
+        list: List of ip_addresses for all pods whose name contains
+              the text in pod type
+
+    """
+    ipaddrs = get_ip_addrs(namespace)
+    return [ipaddrs[cpod] for cpod in ipaddrs if pod_type in cpod]
+
+
+def get_ip_addrs(namespace='openshift-storage'):
+    """
+    Get the ip addresses of all pods in the namespaces.
+
+    Args:
+        namespace (str): namespace in -n field of oc commands
+                         Defaults to ROOK_CLUSTER_NAMESPACE
+
+    Returns:
+        dictionary of ip_address values indexed by pod_name.
+
+    """
+    ret_value = {}
+    client = ac.get_api_client("OCRESTClient")
+    for pod in client.get_pods(namespace=namespace):
```

Comment:
This will call for getting pods every time you iterate.
Please save the value returned from the get and iterate over it

Reply:
Done